bumble/hci.py

```diff
@@ -2302,6 +2325,14 @@ def inner(cls: type[Command]) -> type[Command]:
 
         return inner
 
+    @classmethod
+    def dataclass_command(cls, subclass):
+        # TODO: Move this to __post_init__ when all packets become dataclasses.
+        subclass.parameters = property(
```

Comment:
The reason why `parameters` is a static field, and not computed on the fly is that when an HCI object is created from bytes, the `parameters` byte array is already known, so it can the stored as-is, and therefore doesn't have to be re-serialized when the same object is converted back to bytes. When an HCI object is created from fields, `parameters` is obtained by serialization. Since it is almost always the case that an HCI object will be either created from bytes, or serialized to bytes, or both (for example in the HCI bridge case), this sort of works well. It could be improved by only computing the `parameters` bytes lazily when the object is converted to bytes, in case someone has a use for an HCI object that is created from fields and never serialized

Reply:
Indeed, parameters should be evaluated statically, but currently if we want to keep both legacy and class-based packet constructors, there isn't a simple way to keep both working as is, since __post_init__ in a non-dataclass are ignored in derived dataclasses.

When finally all classes have become dataclasses, things will be simpler. Here we can set a functools.cache property to prevent repeated serialization.